docs: add deepagents v0.7.0 alpha changelog entry (#4771)
langchain-ai/docs: src/oss/python/releases/changelog.mdx

</Callout>

<Update label="Jul 7, 2026" tags={["deepagents"]} rss={{ title: "Jul 7, 2026 - deepagents" }}>
    ## `deepagents` v0.7.0

    - **New [`delete`](/oss/deepagents/tools#built-in-harness-tools) filesystem tool**: Delete a file, or recursively delete a directory and its contents. Backends that don't support deletion have the tool automatically hidden from the model.
    - **`write_file` now overwrites existing files**: `write_file` used to error if the target file already existed. It now overwrites it — use `edit_file` for targeted changes to an existing file.
    - **[Override a default middleware instance](/oss/deepagents/customization#override-a-default-middleware-instance)**: A `middleware=` (or subagent `middleware`) instance whose `.name` matches a default now replaces that default in place, instead of erroring on duplicate middleware.
    - **[Restrict filesystem tools](/oss/deepagents/overview#virtual-filesystem-access)**: `FilesystemMiddleware` now accepts a `tools` allowlist to expose only a subset of the built-in filesystem tools to the model, building on the middleware-override behavior above.

>     Python only — the `deepagents` JS SDK has not adopted these changes yet.

Reviewer: ```suggestion 
<Note> 
The above changes are available only in the `deepagents` Python SDK.
</Note>
```
